### PR TITLE
[#135838391] returning null if accessing peer agent public API before peer agent initialization

### PR DESCRIPTION
**Pull request by ruslandinov**
PT story https://www.pivotaltracker.com/n/projects/1597593/stories/135838391

### CODE REVIEW

File: lib/hlsjs-p2p-wrapper.js [hunk 1]
```diff
@@ -9,7 +9,12 @@ class HlsjsP2PWrapper {
     constructor(p2pConfig, hlsjs) {
         let wrapper = new HlsjsP2PWrapperPrivate(p2pConfig, hlsjs, StreamrootPeerAgentModule);
 
-        this.peerAgent = PeerAgentAPI.get(wrapper);
+        let peerAgentAPI = PeerAgentAPI.get(wrapper);
+        Object.defineProperty(this, 'peerAgent', {
+            get() {
+                return wrapper.peerAgentModule ? peerAgentAPI : null;
+            }
+        });
```
Comment (AxelDelmas) [inline lib/hlsjs-p2p-wrapper.js]: I'd rather use an ES6 getter, but I guess you didn't want to have to store `this._wrapper` ?

**Comment by ruslandinov** [comment]
Exactly

File: test/html/wrapper.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import HlsjsP2PWrapper from "../../lib/hlsjs-p2p-wrapper.js";
+import Hls from "hls.js";
+import should from 'should';
+
+describe("HlsjsP2PWrapper", function() { // using plain ES5 function here
+                                            // otherwise `this.timeout` is broken
+    this.timeout(10000);
```
Comment (jlouazel) [inline test/html/wrapper.js]: Except if you declare it in the **it** scope if I remember it well.

Comment (ruslandinov) [inline test/html/wrapper.js]: Hm... Didn't know that, it was a copypasted legacy from Stephan's tests, I didn't check if it's still valid due to urgency of the fix. 

But that's a good idea to check it and improve the testies if you remember right :)

**Pull request merged**